### ISSUE TITLE
How validate functions passed in objects?

### ISSUE THREAD

**Issue by diosney** [typestack/class-validator#317]
HI, I have as input params an object like this:

```
   hooks: {
          onCreatePayment?: IHookOnCreatePayment;
          onExecutedPayment?: IHookOnExecutedPayment;
          onClick?: IHookOnClick;
          onCancel?: IHookOnCancel;
          onError?: IHookOnError;
   }

export interface IHookOnCancel {
  (): Promise<void>;
}

export interface IHookOnError {
  (error: IError | Error): Promise<void>;
}

export interface IHookOnClick {
  (): Promise<void>;
}

export interface IHookOnCreatePayment {
  (paymentService: PaymentService): Promise<IPayment>;
}

export interface IHookOnExecutedPayment {
  (): Promise<void>;
}
```

There is possible to validate that structure with this library? 

I already read that validations on method arguments is not possible yet but it is planned in the Roadmap, but some other workaround is feasible to do right now?

Thanks

**Comment by diosney** [comment]
I ended up doing something like the following which checks to definition but I will have to wait until function parameters can be validated:

```
export class InitialConfigHooks implements Required<IButtonHooksMap> {
  @IsDefined()
  onCreatePayment: IHookOnCreatePayment;

  @IsDefined()
  onExecutedPayment: IHookOnExecutedPayment;

  @IsDefined()
  onClick: IHookOnClick;

  @IsDefined()
  onCancel: IHookOnCancel;

  @IsDefined()
  onError: IHookOnError;

  constructor(config: Required<IButtonHooksMap>) {
    this.onCreatePayment   = config.onCreatePayment;
    this.onExecutedPayment = config.onExecutedPayment;
    this.onClick           = config.onClick;
    this.onCancel          = config.onCancel;
    this.onError           = config.onError;
  }
}
```

**Comment by 19majkel94** [comment]
I guess class-validator is designed for data only.
Functions can't be passed by JSON so they exist in runtime and you can use TypeScript to check if they have compatible signature.

**Comment by diosney** [comment]
@19majkel94 No problem, you are right, thanks for clarify me that, I guess I have to do it manually using another method, despite that, this library is awesome, keep the hard work :) 

Thanks!

**Comment by diosney** [comment]
Sadly, using Typescript for runtime interface checks are not possible, since they are stripped at compile time, so I have to find another way.

I tried already to integrate https://github.com/gristlabs/ts-interface-builder but have severe limitations and I had to change library, that was how I stumbled upon this.

Nevertheless, as I said, this `class-validator` library is awesome to validate data, so keep the hard work!

Thanks!
